### PR TITLE
Fixes Bug 907499 - added checksum to collector

### PR DESCRIPTION
**Pull request by twobraids**
I've benchmarked for two days and can detect no impact on the collectors whats so ever.  Perhaps we're being overly cautious in not wanting to put the md5sum code into the collector.  This is clearly the most logica place to put it.

### CODE REVIEW

**Comment by rhelmer** [comment]
I don't think there's any harm in making the collector do a checksum of the already in-memory minidump and put that into S3. The part I was more worried about is whether collector would be checking for duplicates and use that to decide whether or not to queue the crash for processing.

I guess we could use S3 for this - just check for the presence of a key with the md5sum for the name - it wouldn't be perfect due to eventual consistency, but it'd probably catch a fair number of them.

If we deferred the duplicate checking to processors they could do this in Postgres instead, which would be totally consistent, but I don't think we'd want to hook collectors up to Postgres at all.

**Comment by twobraids** [comment]
that is the same conclusion that I have drawn.  calculate in the collector - do something with the data elsewhere.

**Comment by rhelmer** [comment]
Another option for consistency is DynamoDB which has (non-default) option for strongly consistent reads.

**Comment by selenamarie** [comment]
@twobraids @rhelmer i concur!

The last bit of IRC conversation led me to: 
```
collector (md5) --> KV store (memcache? redis? aerospikeDB?) <-- processors
     |                                                                  |
     |                                                                  |
     --------------> raw_crash incl md5  <------------------------------
```

**Comment by selenamarie** [comment]
Can we just do this? :checkered_flag: 

I'm super excited to end that reports_duplicates job :)

File: socorro/unittest/collector/test_wsgi_breakpad_collector.py [hunk 1]
```diff
@@ -64,7 +67,11 @@ def test_make_raw_crash(self):
         eq_(rc.some_field, '23')
         eq_(rc.some_other_field, 'XYZ')
 
-    def test_POST(self):
+    @mock.patch('socorro.collector.wsgi_breakpad_collector.time')
```
Comment (selenamarie) [inline socorro/unittest/collector/test_wsgi_breakpad_collector.py]: I like this change to use decorators!  Much more readable.

**Comment by selenamarie** [comment]
r+ i dig the example of using a standard lib with `class_converter` in this patch!  And thanks again for refactoring those tests.

**Pull request merged**